### ISSUE TITLE
[Feature][Connector-V2] Need help for supporting multi-table source feature

### ISSUE THREAD

**Issue by hailin0** [apache/seatunnel#5677]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

After #5395,#5581 finnished. SeaTunnel source can implement multi-table read for zeta. We should add implement multi-table interface for all SeaTunnel source, also need e2e testcase & docs. Please refer Jdbc source from #5581.

## Please pick connector you want to implement it!

| Type | Connector | Contributer | Status | Issue/PR |
| ---- | ---- | ---- | ---- | ---- |
| Source | fakesource | | Done | | 
| Source | jdbc | | Done |#5581 |
| Source | Local-file |@ruanwenjun | Done | https://github.com/apache/seatunnel/pull/5781|   
| Source | cdc-mysql | |Done | |  
| Source | cdc-sqlserver | | Done| | 
| Source | cdc-oracle | | Done| | 
| Source | cdc-mongodb |@chl-wxp |Doing| | 
| Source | tdengine | | | |  
| Source | starrocks | |Done | |   
| Source | socket | | | | 
| Source | paimon | | | |  
| Source | openmldb | | | |  
| Source | neo4j | | | | 
| Source | mongodb |@chl-wxp | Doing| |  
| Source | maxcompute | | Done| |  
| Source | kudu | @Carl-Zhou-CN | Done |https://github.com/apache/seatunnel/issues/5714 |  
| Source | kafka | @zhilinli123|  | https://github.com/apache/seatunnel/pull/5992|  
| Source | pulsar | | | |  
| Source | iotdb | | | |   
| Source | influxdb | | | |  
| Source | elasticsearch | @FuYouJ | | |  
| Source | iceberg |@LiJie20190102| | |  
| Source | hudi |@LiJie20190102 | | |  
| Source | hive |@ruanwenjun | | https://github.com/apache/seatunnel/pull/5929|  
| Source | localfile |@ruanwenjun | Done| https://github.com/apache/seatunnel/pull/5781|  
| Source | clickhouse |  | | https://github.com/apache/seatunnel/issues/6230| 
| Source | cassandra | | | |  
| Source | amazondynamodb | | | | 
| Source | Doris | @happyboy1024 |Done | https://github.com/apache/seatunnel/issues/5731| 

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by hailin0** [comment]
link https://github.com/apache/seatunnel/issues/5652

**Comment by Carl-Zhou-CN** [comment]
Please assign me kudu

**Comment by hailin0** [comment]
> Please assign me kudu

@Carl-Zhou-CN 
Please comment this issue
https://github.com/apache/seatunnel/issues/5714

**Comment by FlechazoW** [comment]
paimon\hive\cassandra please assign to me.

**Comment by hailin0** [comment]
> paimon\hive\cassandra please assign to me.

@FlechazoW 
I suggest you develop a connector first and wait until you commit the merge before continuing

**Comment by CodingGPT** [comment]
doris  please assign to me, tks

**Comment by hailin0** [comment]
> [Feature][Kudu] Support multi-table source read #5714

@CodingGPT Please comment this issue

https://github.com/apache/seatunnel/issues/5731

**Comment by Carl-Zhou-CN** [comment]
clickhouse please assign to me, tks

**Comment by zhilinli123** [comment]
kafka multi-source , temporarily assigned to me and I will try to complete it,thks

**Comment by hailin0** [comment]
> [Feature][Kafka] Support multi-table source read #5992

@Carl-Zhou-CN 
https://github.com/apache/seatunnel/issues/6230

**Comment by FuYouJ** [comment]
es please assign to me,

**Comment by zhilinli123** [comment]
paimon Temporarily assigned to me thks

**Comment by 531651225** [comment]
influxdb please assign to me

**Comment by xiaochen-zhou** [comment]
starrocks please assign to me and I will try to complete it , thx

**Comment by hailin0** [comment]
> es please assign to me,

@FuYouJ Please create new es issue link to this

**Comment by hailin0** [comment]
> Please create new es issue link to this

@zhilinli123 
Please create new issue link to this

**Comment by hailin0** [comment]
> influxdb please assign to me

@531651225 
Please create new issue link to this

**Comment by hailin0** [comment]
> starrocks please assign to me and I will try to complete it , thx

@xiaochen-zhou 
Please create new issue link to this

**Comment by FuYouJ** [comment]
es support multi-table source feature
https://github.com/apache/seatunnel/issues/6730

**Comment by 531651225** [comment]
> > influxdb please assign to me
> 
> @531651225 Please create new issue link to this

[Feature][influxdb] Support multi-table source feature [#6925](https://github.com/apache/seatunnel/issues/6925)

**Comment by davidzollo** [comment]
If there is anybody who'd like to implement these connector features, please leave a related message like `I'd like to implement XX multi-table source` and then you can submit your PR within four weeks. 

A very kindly reminder: Other guys can pick it up if the related PR is not submitted within four weeks. ^-^

Here are some helpful resources to get started:  
- [Contribution Setup Guide](https://seatunnel.apache.org/docs/contribution/setup)  
- [Code Submission Guide](https://seatunnel.apache.org/community/submit_guide/submit-code)  

By the way, since this might be your first contribution, I think we can have a more in-depth communication to help you get familiar with SeaTunnel. Feel free to contact me on LinkedIn ([David Zollo](https://www.linkedin.com/in/davidzollo)) or WeChat ID(davidzollo). When adding me, please let me know your GitHub ID. I'd be happy to assist you!

**Comment by davidzollo** [comment]
I've reorganized this list, and there are still many connectors to be implemented here.
If there is anybody who'd like to implement these connector features, please leave a related message like `I'd like to implement XX multi-table source` and then you can submit your PR within four weeks. 

A very kindly reminder: Other guys can pick it up if the related PR is not submitted within four weeks. ^-^

Here are some helpful resources to get started:  
- [Contribution Setup Guide](https://seatunnel.apache.org/docs/contribution/setup)  
- [Code Submission Guide](https://seatunnel.apache.org/community/submit_guide/submit-code)  

By the way, since this might be your first contribution, I think we can have a more in-depth communication to help you get familiar with SeaTunnel. Feel free to contact me on LinkedIn ([David Zollo](https://www.linkedin.com/in/davidzollo)) or WeChat ID(davidzollo). When adding me, please let me know your GitHub ID. I'd be happy to assist you!

**Comment by davidzollo** [comment]
Please leave a message if anybody would like to implement the multi - table source feature for any connector.

**Comment by LiJie20190102** [comment]
Please assign hudi/iceberg to me, thx

**Comment by davidzollo** [comment]
> Please assign hudi/iceberg to me, thx

Good, I've assigned to you and looking forward to your PRs

**Comment by davidzollo** [comment]
I have newly organized a new issue to continue tracking the progress of this list. [[Feature][Connector] Need help for supporting multi-table source feature #10425](https://github.com/apache/seatunnel/issues/10425)
